### PR TITLE
Change data context in the IPC calls for relocation.

### PR DESCRIPTION
**Pull request by mamokarz**

### CODE REVIEW

File: deps/azure-core-c/CMakeLists.txt [hunk 1]
```diff
@@ -57,22 +99,26 @@ add_library (
...
-target_include_directories (az_core
+target_include_directories (az_core_pic
   PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/inc>
   $<INSTALL_INTERFACE:include/az_core>
 )
 
+target_compile_options(az_core_pic
+    PRIVATE
+        -fpie -fno-plt -fno-jump-tables ${PIC_FLAGS})
```
Comment (mamokarz) [inline deps/azure-core-c/CMakeLists.txt:110]: >-fno-plt [](http://example.com/codeflow?start=14&length=8)

Old Ubuntu in the gate didn't like it. Move to PIC_FLAGS.

File: pal/GCC/ARM_CM4F/az_ulib_port.h [hunk 2]
```diff
@@ -87,6 +87,18 @@ extern "C"
...
   }
 
+  __attribute__((always_inline)) static inline void AZ_ULIB_PORT_GET_DATA_CONTEXT(
+      volatile void** data_address)
+  {
+    __asm inline("       str     r9, [%0]                " : "=&r"(data_address) : : "memory");
+  }
+
+  __attribute__((always_inline)) static inline void AZ_ULIB_PORT_SET_DATA_CONTEXT(
+      volatile void* data_address)
```
Comment (smcd253) [inline pal/GCC/ARM_CM4F/az_ulib_port.h:97]: are these only implemented for cortex m4?

Comment (mamokarz) [reply pal/GCC/ARM_CM4F/az_ulib_port.h:97]: Yes, Windows and Linux handle relocation in a different way, using the MMU.

**Review by smcd253 — APPROVED** [review]
lgtm

**Pull request merged**